packages/codec/lib/compilations/utils.ts

```diff
@@ -4,16 +4,54 @@ const debug = debugModule("codec:compilations:utils");
 import * as Ast from "@truffle/codec/ast";
 import * as Compiler from "@truffle/codec/compiler";
 import { ContractObject as Artifact } from "@truffle/contract-schema/spec";
-import { CompiledContract } from "@truffle/compile-common";
+import {
+  CompiledContract,
+  Compilation as CompilerCompilation
+} from "@truffle/compile-common";
 import { Compilation, Contract, Source } from "./types";
 
+export function shimCompilations(
+  inputCompilations: CompilerCompilation[],
+  shimmedCompilationIdPrefix = "shimmedcompilation",
+  externalSolidity = false
+): Compilation[] {
+  return inputCompilations.map(
+    ({ contracts, sourceIndexes }, compilationIndex) =>
+      shimContracts(
+        contracts,
+        sourceIndexes,
+        `${shimmedCompilationIdPrefix}Number(${compilationIndex})`,
+        externalSolidity
+      )
+  );
+}
+
+/**
+ * wrapper around shimArtifactsToCompilation that just returns
+ * the result in a one-element array (keeping the old name
+ * shimArtifacts for compatibility)
+ */
 export function shimArtifacts(
```

Comment:
Do we still use this anywhere?

Reply:
Oh right Teams might use it. @kevinbluer @seesemichaelj are you using this function? Can you switch to using the new one here, `shimCompilations`?

Reply:
Yup, it's used multiple times here: https://github.com/trufflesuite/teams/blob/develop/packages/api/src/schema/debugger/resolvers.ts#L93-L117

I don't see a reason why we can't switch unless the functionality is completely changing. I'll leave this one up to @kevinbluer

Reply:
Thanks for the response @seesemichaelj! Looking at usage, it does seem to be fine - you already have multiple compilations and you're compressing them just for `shimArtifacts`. This would remove the extra step.

Reply:
The debugger CLI uses it as well when actually running off of artifacts.  I've changed it to use `shimCompilations` only in the context where it's using compiler output, not artifacts.